torch/_inductor/scheduler.py

```diff
@@ -205,29 +292,40 @@ def prune_deps(self) -> None:
     def prune_weak_deps(self) -> None:
         # Prune weak dependencies on buffers that have been removed
         def should_prune(dep: Dep) -> bool:
-            return isinstance(dep, WeakDep) and dep.name in V.graph.removed_buffers
+            if not isinstance(dep, WeakDep):
+                return False
+            op = self.scheduler.name_to_buf[dep.name].defining_op
+            return all(
+                buf.get_name() in V.graph.removed_buffers for buf in op.get_outputs()
+            )
```

Comment:
why are you lifting the pruning from a buffer level to an operation level? From `WeakDep`:
```
# Used for tracking mutation ordering
# if A reads a buffer and B mutates it
# B must be ordered after A
```
Weak deps act at a buffer level, not at an op level?

Reply:
The weak dep is really depending on the read that occurs within the operation, whereas the buffer represents the write. So it really is the operation that we depend on here.

That being said, I should probably change this to use `removed_operations`.

Reply:
Changed